### PR TITLE
Migrate builds from Travis CI to GitHub actions

### PR DESCRIPTION
**Pull request by jakzal**
image

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
You may as well remove the travis config as part of the PR if it's comprehensively replaced

**Comment by jakzal** [comment]
@ciaranmcnulty I did...

**Comment by ciaranmcnulty** [comment]
@jakzal was confused why it looked like it was running but it’s a required check for the master branch

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                run: ./bin/behat -fprogress --strict $([ -z "${{ matrix.skip-interop }}" ] || echo "--tags ~@interop&&~@php8")
+
+            -   name: Run tests (Behat for PHP 8.0)
+                if: matrix.php >= 8.0
+                run: ./bin/behat -fprogress --strict --tags=@php8
+
+            -   name: Build the PHAR
+                if: matrix.php != '8.0'
+                run: |
+                    curl -LSs https://box-project.github.io/box2/installer.php | php &&
```
Comment (smoench) [inline .github/workflows/build.yml:64]: This downloads a very old version of box. As far as I understand that script it downloads one of these promoted files https://box-project.github.io/box2/manifest.json.

Comment (jakzal) [reply .github/workflows/build.yml:64]: Yes, my approach is to migrate everything as is and do any updates as a separate step.

Behat still uses box1. I think @ciaranmcnulty would appreciate if someone upgraded it and added phar scoping ;)

Comment (stof) [reply .github/workflows/build.yml:64]: I don't think we can scope phars: behat extensions have to deal with Symfony components, not only with the Behat code itself.

**Review by pamil — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (jakzal) [reply .github/workflows/build.yml:64]: This only means we can't scope PHARs fully (particularly in places where the contract of a Symfony component is exposed to the end-users).

Comment (stof) [reply .github/workflows/build.yml:64]: if we don't scope Behat packages and we don't scope Symfony components, there isn't much left to scope: https://github.com/Behat/Behat/blob/304661d4bc00934314c835c79fe36c0ce38da268/composer.json#L16-L28

Comment (ciaranmcnulty) [reply .github/workflows/build.yml:64]: Hm, I'd seen phar scoping as a good reason for dropping older symfony versions

**Pull request merged**